src/orion/executor/multiprocess_backend.py

```diff
@@ -0,0 +1,209 @@
+import dataclasses
+import logging
+import pickle
+import traceback
+import uuid
+from concurrent.futures import ThreadPoolExecutor, TimeoutError, wait
+from dataclasses import dataclass
+from multiprocessing import Manager, Process, get_context
+from multiprocessing.pool import AsyncResult
+from multiprocessing.pool import Pool as PyPool
+from queue import Empty
+
+import cloudpickle
+
+from orion.executor.base import AsyncException, AsyncResult, BaseExecutor
+
+log = logging.getLogger(__name__)
+
+
+def _couldpickle_exec(payload):
+    function, args, kwargs = pickle.loads(payload)
+    result = function(*args, **kwargs)
+    return cloudpickle.dumps(result)
+
+
+class _Process(Process):
+    """Process that cannot be a daemon"""
+
+    def _get_daemon(self):
+        return False
+
+    def _set_daemon(self, value):
+        pass
+
+    daemon = property(_get_daemon, _set_daemon)
+
+
+class _Future:
+    """Wraps a python AsyncResult"""
+
+    def __init__(self, future, cloudpickle=False):
+        self.future = future
+        self.cloudpickle = cloudpickle
+
+    def get(self, timeout=None):
+        r = self.future.get(timeout)
+        return pickle.loads(r) if self.cloudpickle else r
+
+    def wait(self, timeout=None):
+        return self.future.wait(timeout)
```

Comment:
Is the future object here expected to be a `concurrent.futures.Future`? I cannot find the method wait in the docs: https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future

Reply:
* ``_Future`` is https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.AsyncResult.wait

* ``_ThreadFuture`` is https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future

Reply:
Documentation of the class should explain what differs from multiprocessing.pool.AsyncResult. I know the code is simple but users should be able to understand this difference from doc directly.

Reply:
Why should they understand ? The futures are private within Orion, the differences between AsyncResult and Future are implementation details, that is why we have an adapter that unify the interfaces.

Reply:
This part is indeed not at the top layer but it is exposed and available if users also want to use executors for processing other things than Oríon trials.

Reply:
Still not sure what you want me to do about it.
The Future Interface is documented, all the adapters inherit that doc and the adapter mention explicitly which flavor of future they are adapting

Reply:
There are no differences except for cloudpickle right?

Reply:
yeah

Reply:
So that's the only thing I see that would need to be documented. It would be great to know why it's needed.

Reply:
added the comment, it is to have feature parity with Dask, so things that can be pickled with one can be pickled with the other